example/sysinfo/odp_sysinfo.c

```diff
@@ -25,52 +25,206 @@ static const char *support_level(odp_support_t support)
+		return "aes_gcm";
+	case ODP_AUTH_ALG_AES_GMAC:
+		return "aes_gmac";
+	case ODP_AUTH_ALG_AES_CCM:
+		return "aes_ccm";
+	case ODP_AUTH_ALG_AES_CMAC:
+		return "aes_cmac";
+	case ODP_AUTH_ALG_CHACHA20_POLY1305:
+		return "chacha20_poly1305";
+	default:
+		return "Unknown";
+	}
+}
+
 static void print_cipher_algos(odp_crypto_cipher_algos_t ciphers)
 {
 	if (ciphers.bit.null)
-		printf("null ");
+		printf("%s ", cipher_alg_name(ODP_CIPHER_ALG_NULL));
 	if (ciphers.bit.des)
-		printf("des ");
+		printf("%s ", cipher_alg_name(ODP_CIPHER_ALG_DES));
 	if (ciphers.bit.trides_cbc)
-		printf("trides_cbc ");
+		printf("%s ", cipher_alg_name(ODP_CIPHER_ALG_3DES_CBC));
 	if (ciphers.bit.aes_cbc)
-		printf("aes_cbc ");
+		printf("%s ", cipher_alg_name(ODP_CIPHER_ALG_AES_CBC));
 	if (ciphers.bit.aes_ctr)
-		printf("aes_ctr ");
+		printf("%s ", cipher_alg_name(ODP_CIPHER_ALG_AES_CTR));
 	if (ciphers.bit.aes_gcm)
-		printf("aes_gcm ");
+		printf("%s ", cipher_alg_name(ODP_CIPHER_ALG_AES_GCM));
 	if (ciphers.bit.aes_ccm)
-		printf("aes_ccm ");
+		printf("%s ", cipher_alg_name(ODP_CIPHER_ALG_AES_CCM));
 	if (ciphers.bit.chacha20_poly1305)
-		printf("chacha20_poly1305 ");
+		printf("%s ",
+		       cipher_alg_name(ODP_CIPHER_ALG_CHACHA20_POLY1305));
 }
 
 static void print_auth_algos(odp_crypto_auth_algos_t auths)
 {
 	if (auths.bit.null)
-		printf("null ");
+		printf("%s ", auth_alg_name(ODP_AUTH_ALG_NULL));
+	if (auths.bit.md5_hmac)
+		printf("%s ", auth_alg_name(ODP_AUTH_ALG_MD5_HMAC));
+	if (auths.bit.sha1_hmac)
+		printf("%s ", auth_alg_name(ODP_AUTH_ALG_SHA1_HMAC));
+	if (auths.bit.sha256_hmac)
+		printf("%s ", auth_alg_name(ODP_AUTH_ALG_SHA256_HMAC));
+	if (auths.bit.sha384_hmac)
+		printf("%s ", auth_alg_name(ODP_AUTH_ALG_SHA384_HMAC));
+	if (auths.bit.sha512_hmac)
+		printf("%s ", auth_alg_name(ODP_AUTH_ALG_SHA512_HMAC));
+	if (auths.bit.aes_gcm)
+		printf("%s ", auth_alg_name(ODP_AUTH_ALG_AES_GCM));
+	if (auths.bit.aes_gmac)
+		printf("%s ", auth_alg_name(ODP_AUTH_ALG_AES_GMAC));
+	if (auths.bit.aes_ccm)
+		printf("%s ", auth_alg_name(ODP_AUTH_ALG_AES_CCM));
+	if (auths.bit.aes_cmac)
+		printf("%s ", auth_alg_name(ODP_AUTH_ALG_AES_CMAC));
+	if (auths.bit.aes_xcbc_mac)
+		printf("%s ", auth_alg_name(ODP_AUTH_ALG_AES_XCBC_MAC));
+	if (auths.bit.chacha20_poly1305)
+		printf("%s ", auth_alg_name(ODP_AUTH_ALG_CHACHA20_POLY1305));
+}
+
+static void print_cipher_capa(odp_cipher_alg_t cipher)
+{
+	int caps = odp_crypto_cipher_capability(cipher, NULL, 0);
+	int rc, i;
+
+	if (caps < 0)
+		return;
+
+	odp_crypto_cipher_capability_t capa[caps];
+
+	rc = odp_crypto_cipher_capability(cipher, capa, caps);
+	if (rc < 0)
+		return;
+
+	printf("        %s:\n", cipher_alg_name(cipher));
+	for (i = 0; i < rc; i++)
+		printf("            key %d iv %d\n",
+		       capa[i].key_len, capa[i].iv_len);
+}
+
+static void print_auth_capa(odp_auth_alg_t auth)
+{
+	int caps = odp_crypto_auth_capability(auth, NULL, 0);
+	int rc, i;
+
+	if (caps < 0)
+		return;
```

Comment:
Similar to earlier comment, `if (caps <= 0)` here.

Reply:
ack